datamade/scan-adapter
# Add simple ETL pipeline to reformat data, connects #1

## Description

See title. I'll set up a scheduled run separately. You can view the processed data, here: https://docs.google.com/spreadsheets/d/1j-bccaWHtag0B5rFjDUMkUPShqNftpD_7wwfwqpZktg/edit?usp=sharing

(Google Sheets added an apostrophe before each measurement that is not present in the actual output, IDK what's up with that.)

h/t @jeancochrane for the YAML inspiration ❤️

### Notes

- I added all genes currently being tested for. One downside to this approach is that new genes will need to be added manually, but on the other hand, adding (or removing) a gene simply entails an update to the `targets.yml` file.
  - If the SCAN team provides human friendly names, we can add them to the currently empty verbose name keys and use that in the output, instead. 
- I wasn't sure whether we wanted to include records for genes that were not tested for on a particular day (blank results), so I went with yes. Omitting would be super easy.
- There are a couple of genes with normalized values without confidence intervals. Wasn't sure what to do with those, so I omitted those measurements. Can the SCAN team clarify?
- I added source fields for each of the measurements for a clearer audit trail and easier debugging.

### Testing instructions

- Pull down this branch, then follow the instructions in the README to make the processed data.
- Review the data and confirm it looks like you'd expect. 

Connects #1.
